.github/workflows/website.yml

```
@@ -48,30 +68,77 @@ jobs:
       - uses: bahmutov/npm-install@v1
         env:
           PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
-      - run: npm run build -w packages/website
+      - run: touch .env
+      - run: |
+          (
+            npm -w '@web3-storage/website' run ci:envfile --silent \
+            | sed "s/^NEXT_PUBLIC_MAGIC=$/NEXT_PUBLIC_MAGIC=$NEXT_PUBLIC_MAGIC/" \
+            | sed "s/^MAGIC_SECRET_KEY=$/MAGIC_SECRET_KEY=$MAGIC_SECRET_KEY/" \
+            ;
+            echo 'NEXT_PUBLIC_MAGIC_TESTMODE_ENABLED="1"';
+          ) > .env
         env:
-          NEXT_PUBLIC_API: http://localhost:8787
-          NEXT_PUBLIC_ENV: test-e2e
           NEXT_PUBLIC_MAGIC: ${{ secrets.STAGING_MAGIC_PUBLIC_KEY }}
+          MAGIC_SECRET_KEY: ${{ secrets.STAGING_MAGIC_SECRET_KEY }}
+      - name: load db schema
+        run: |
+          npm -w '@web3-storage/db' run start
+          npm -w '@web3-storage/db' run load-schema
+          npm -w '@web3-storage/db' run stop
+      - run: node -r dotenv/config $(which npm) run build -w packages/website
```

What happens if we specify the `env` on this step? That should work. If it doesn't we should fix that.
```suggestion
      - run: npm run build -w packages/website
        env:
          NEXT_PUBLIC_MAGIC: ...
```

---

it wont work! lol 🤣 

So this is another quirk of how our npm scripts sorta rely on dotenv.

`npm run build -w packages/website` will run the build script in that package, which will use dotenv. BUT. dotenv will read from `.env` within the package website, not within the monorepo root.

This workflow only writes it to the monorepo root. And then we want those envs to be used by more than just the website package (e.g. also api, db)

---

If we can, lets fix the bug that is _(how we set env vars for building the website build)_

---

Turns out we dont need to do a (prod) build of the website here because later when playwright runs (monorepo) `npm start` , it'll run `start` for api, which will run miniflare in such a way that it triggers a build based on the env of that step. https://github.com/web3-storage/web3.storage/pull/1782#issuecomment-1223363343